### ISSUE TITLE
Is RevIN causal?

### ISSUE THREAD

**Issue by jaak-s** [ts-kim/RevIN#15]
Thank you for the code.

I did a quick check with revin to see if it is causal along the sequence length axis, where by "causal" I mean that only the past and current tokens are used for transforming the token (or in other words that the future tokens cannot impact past). My results show that RevIN is **not** causal.

Can you confirm that this is correct? Thanks!

**Comment by Shangshi-Chen** [comment]
You are right. It's not causal. To make it so, you can truncate x into only historical data in the _get_statistics method.
